From 6c6e9c82a431194f49c0500b000fc19bf3ad2a6f Mon Sep 17 00:00:00 2001
From: Jonathan Richard Robert Kimmitt <jrrk2@cl.cam.ac.uk>
Date: Tue, 14 May 2019 15:43:56 +0100
Subject: [PATCH 3/7] Revert fp dirty hacks

---
 arch/riscv/include/asm/switch_to.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/arch/riscv/include/asm/switch_to.h b/arch/riscv/include/asm/switch_to.h
index 322fefd9d296..6291095b363a 100644
--- a/arch/riscv/include/asm/switch_to.h
+++ b/arch/riscv/include/asm/switch_to.h
@@ -29,12 +29,7 @@ static inline void fstate_off(struct task_struct *task,
 static inline void fstate_save(struct task_struct *task,
 			       struct pt_regs *regs)
 {
-  // JRRK: Hack around buggy support for SR_FS and/or SR_SD
-#if 0  
 	if ((regs->sstatus & SR_FS) == SR_FS_DIRTY) {
-#else          
-	if ((regs->sstatus & SR_FS) != SR_FS_OFF) {
-#endif          
 		__fstate_save(task);
 		__fstate_clean(regs);
 	}
@@ -55,10 +50,7 @@ static inline void __switch_to_aux(struct task_struct *prev,
 	struct pt_regs *regs;
 
 	regs = task_pt_regs(prev);
-  // JRRK: Hack around buggy support for SR_FS and/or SR_SD
-#if 0  
 	if (unlikely(regs->sstatus & SR_SD))
-#endif          
 		fstate_save(prev, regs);
 	fstate_restore(next, task_pt_regs(next));
 }
-- 
2.25.1

