From 5bc340654be875e3df1cbe9f77c6b4c38180a6c8 Mon Sep 17 00:00:00 2001
From: Jonathan Richard Robert Kimmitt <jrrk2@cl.cam.ac.uk>
Date: Wed, 15 May 2019 12:21:39 +0100
Subject: [PATCH 5/7] Update .gitignore

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index d01cda8e1177..ea22a43950eb 100644
--- a/.gitignore
+++ b/.gitignore
@@ -155,3 +155,4 @@ x509.genkey
 
 # Documentation toolchain
 sphinx_*/
+initramfs.cpio
-- 
2.25.1

